### ISSUE TITLE
CamelConnector init should check for existing CamelReactiveStreamsService configuration

### ISSUE THREAD

**Issue by jamesnetherton** [smallrye/smallrye-reactive-messaging#781]
It's possible that the `CamelReactiveStreamsService` is already configured on the `CamelContext` before `CamelConnector` attempts its own configuration in the `@PostConstruct` annotation `init` method.

This is the case in Camel Quarkus where its reactive streams extension has already pre-configured `CamelReactiveStreamsService`.

It would be good if `CamelConnector.init()` had a check like:

```java
@PostConstruct
public void init() {
    reactive = CamelReactiveStreams.get(camel);
    if (reactive == null) {
        // Configuration logic here...
    }
}
```

**Comment by cescoffier** [comment]
+1, let me add that!

**Comment by cescoffier** [comment]
@jamesnetherton I picked a different approach as your code was triggering a lookup loop (was fun to debug). Can you have a look at https://github.com/smallrye/smallrye-reactive-messaging/pull/782 and tell me if that's ok?

**Comment by jamesnetherton** [comment]
@cescoffier Yes, that works for me :+1:

**Comment by jamesnetherton** [comment]
@cescoffier any ideas when we're likely to see 2.4.0 released?

**Comment by cescoffier** [comment]
Tomorrow :-)

The goal is to be included in Quarkus 1.9

Le mar. 29 sept. 2020 à 14:54, James Netherton <notifications@github.com> a
écrit :
